### PR TITLE
Hardware PWM support for the C.H.I.P

### PR DESCRIPTION
**Pull request by erkkah**
This PR adds support for PWM0 on the C.H.I.P.

Second attempt, including the core PWM functionality only, leaving out device tree overlay management.

### CODE REVIEW

**Comment by erkkah** [comment]
Feels like an uphill battle fighting the coverage gods ☺️

**Comment by deadprogram** [comment]
Hi @erkkah this looks very promising. Now we need to give people the ability to install the needed device tree overlay using Gort. :palm_tree:

File: platforms/chip/README.md [hunk 1]
```diff
@@ -11,6 +11,9 @@ For documentation about the C.H.I.P. platform click [here](http://docs.getchip.c
 go get -d -u gobot.io/x/gobot/... && go install gobot.io/x/gobot/platforms/chip
 ```
 
+Note that PWM might not be available in your kernel and you might need to load the right overlays
+to expose PWM on the PWM0 pin.
+
```
Comment (deadprogram) [inline platforms/chip/README.md]: This is where instructions on how to install the dtc overlay on CHIP using Gort should go :)

Comment (erkkah) [inline platforms/chip/README.md]: Done!

**Comment by deadprogram** [comment]
Thanks @erkkah now merging.

**Pull request merged**